docker/build_graphstorm_image.sh

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+trap cleanup SIGINT SIGTERM ERR EXIT
+
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+usage() {
+    cat <<EOF
+Usage: $(basename "${BASH_SOURCE[0]}") [-h] [-x] -e sagemaker
+
+Builds the GraphStorm training/inference Docker images.
+
+Available options:
+
+-h, --help          Print this help and exit
+-x, --verbose       Print script debug info (set -x)
+-e, --environment   Image execution environment. Must be one of 'local' or 'sagemaker'. Required.
+-d, --device        Device type, must be one of 'cpu' or 'gpu'. Default is 'gpu'.
+-p, --path          Path to graphstorm root directory, default is one level above this script's location.
+-i, --image         Docker image name, default is 'graphstorm'.
+-s, --suffix        Suffix for the image tag, can be used to push custom image tags. Default is "<environment>-<device>".
+-b, --build         Docker build directory prefix, default is '/tmp/'.
+--use-parmetis      Include dependencies needed to run ParMETIS distributed partitioning.
```

Comment:
If we support installing parmetis in the docker image, why not copy graphstorm_processing into the docker image in L142-L146?

Reply:
ParMETIS requires the PyTorch image to run, graphstorm_processing requires a PySpark image. 

What's the motivation for copying the GSProcessing code?

Reply:
I'll remove this option and add back once we merge the ParMETIS/local PR.